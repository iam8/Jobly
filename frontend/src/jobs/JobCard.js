import React, {useContext, useState, useEffect} from "react";
import {Button, Card, CardTitle, CardBody} from "reactstrap";

import UserContext from "../auth/UserContext";


/** Show basic information about a job. */
function JobCard({id, title, salary, equity, companyName}) {
    const {hasAppliedToJob, applyToJob} = useContext(UserContext);
    const [hasApplied, setHasApplied] = useState();

    useEffect(() => {

        /** Retrieve the application status of this JobCard. */
        setHasApplied(hasAppliedToJob(id));
    }, [id, hasAppliedToJob]);

    /** Apply for the job associated with this JobCard. */
    const handleApply = async () => {
        if (hasAppliedToJob(id)) return;

        await applyToJob(id);
        setHasApplied(true);
    }

    return (
        <div className="JobCard">
            <Card>
                <CardBody>
                    <CardTitle tag="h5">{title}</CardTitle>

                    {companyName !== undefined &&
                        <p className="text-muted">
                            {companyName}
                        </p>
                    }

                    {salary &&
                            <div><small>Salary: {salary}</small></div>
                    }

                    {equity !== null &&
                            <div><small>Equity: {equity}</small></div>
                    }

                    <Button
                        className="float-end"
                        color="danger"
                        onClick={handleApply}
                        disabled={hasApplied}
                    >
                        {hasApplied ? <span>Applied</span> : <span>Apply</span>}
                    </Button>
                </CardBody>
            </Card>
        </div>
    )
}


export default JobCard;
