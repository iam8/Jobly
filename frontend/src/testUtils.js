import React from "react";
import UserContext from "./auth/UserContext";


const demoUser = {
    username: "testuser",
    firstName: "testfirst",
    lastName: "testlast",
    email: "test@test.net",
};


const DemoUserProvider = ({ children, currentUser = demoUser, hasAppliedToJob = () => false }) => (
    <UserContext.Provider value={{ currentUser, hasAppliedToJob }}>
        {children}
    </UserContext.Provider>
);


export { DemoUserProvider };
