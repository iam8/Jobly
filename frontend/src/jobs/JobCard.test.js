import React from "react";
import { render } from "@testing-library/react";
import { DemoUserProvider as UserProvider } from "../testUtils";
import JobCard from "./JobCard";


const testJob = {
    id: 0,
    title: "Test title",
    salary: 100000,
    equity: 0,
    companyName: "Test Company"
}


test("Renders without crashing", () => {
    render(
        <UserProvider>
            <JobCard
                id={testJob.id}
                title={testJob.title}
                salary={testJob.salary}
                equity={testJob.equity}
                companyName={testJob.companyName}
            />
        </UserProvider>
    );
})


test("Matches snapshot", () => {
    const {asFragment} = render(
        <UserProvider>
            <JobCard
                id={testJob.id}
                title={testJob.title}
                salary={testJob.salary}
                equity={testJob.equity}
                companyName={testJob.companyName}
            />
        </UserProvider>
    );

    expect(asFragment()).toMatchSnapshot();
})