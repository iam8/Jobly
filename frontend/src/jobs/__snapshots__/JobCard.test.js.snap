// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="JobCard"
  >
    <div
      class="card"
    >
      <div
        class="card-body"
      >
        <h5
          class="card-title"
        >
          Test title
        </h5>
        <p
          class="text-muted"
        >
          Test Company
        </p>
        <div>
          <small>
            Salary: 100000
          </small>
        </div>
        <div>
          <small>
            Equity: 0
          </small>
        </div>
        <button
          class="float-end btn btn-danger"
          type="button"
        >
          <span>
            Apply
          </span>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
